src/context.hpp

```diff
@@ -28,7 +33,8 @@ struct cvk_context : public _cl_context,
                      refcounted,
                      object_magic_header<object_magic::context> {
 
-    cvk_context(cvk_device* device, const cl_context_properties* props)
+    cvk_context(cvk_device* device, const cl_context_properties* props,
+                void* user_data)
```

Comment:
Suggest initialising `m_user_data` here. I can't think of a reason to gate the behaviour on the presence of the printf properties.